lib/host/wasi/wasifunc.cpp

```diff
@@ -1848,5 +1848,96 @@ Expect<uint32_t> WasiSockShutdown::body(Runtime::Instance::MemoryInstance *,
   return __WASI_ERRNO_SUCCESS;
 }
 
+Expect<uint32_t>
+WasiGetAddrinfo::body(Runtime::Instance::MemoryInstance *MemInst,
+                      uint32_t NodePtr, uint32_t NodeLen, uint32_t ServicePtr,
+                      uint32_t ServiceLen, uint32_t HintsPtr, uint32_t ResPtr,
+                      uint32_t MaxResLength, uint32_t ResLengthPtr) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  const char *Node = nullptr;
+  if (NodeLen != 0) {
+    Node = MemInst->getPointer<const char *>(NodePtr, NodeLen);
+  }
+  const char *Service = nullptr;
+  if (ServiceLen != 0) {
+    Service = MemInst->getPointer<const char *>(ServicePtr, ServiceLen);
+  }
+  // service and node can not be nullptr at the same time
+  if (Service == nullptr && Node == nullptr) {
+    return __WASI_ERRNO_FAULT;
```

Comment:
According getaddrinfo manual, this error should be `__WASI_ERRNO_AINONAME`
https://man7.org/linux/man-pages/man3/getaddrinfo.3.html